### PR TITLE
Update Arcade and require .NET 6 Preview 7 SDK

### PR DESCRIPTION
**Pull request by JoeRobich**
Arcade is currently pushing early builds of .NET 6 RC which would be difficult to require contributors to install in order to develop against Roslyn. This moves us to the latest published Preview version and brings in recent Arcacde changes.

### CODE REVIEW

**Review by ryzngard — APPROVED** [review]

**Comment by RikkiGibson** [comment]
Do we know why they are pushing such new versions of the SDK?

**Comment by JoeRobich** [comment]
> Do we know why they are pushing such new versions of the SDK?

I have not asked. I assumed it was to get RC as much validation as possible before it gets released.

**Review by jmarolf — APPROVED** [review]

**Comment by jkoritzinsky** [comment]
Arcade moved to the RC1 SDK to help alleviate NuGet feed instability that dotnet/aspnetcore and dotnet/runtime were hitting at extremely high frequency.

**Comment by dibarbet** [comment]
The test CsiScript_WithSourceCodeRedirectedViaStandardInput_ExecutesNonInteractively is failing as well with this update.
```
Compilation error(s) occurred:  System.TypeLoadException: Could not load type 'System.Int32' from assembly 'R*ad9ddb7c-7900-42d3-b5d7-ff74566f9be0#1-0, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null' due to value type mismatch.
  + System.Delegate.BindToMethodInfo(object, System.IRuntimeMethodInfo, System.RuntimeType, System.DelegateBindingFlags)
  + System.Reflection.RuntimeMethodInfo.CreateDelegateInternal(System.Type, object, System.DelegateBindingFlags)
  + System.Reflection.RuntimeMethodInfo.CreateDelegate(System.Type)
  + Roslyn.Utilities.ReflectionUtilities.CreateDelegate<T>(System.Reflection.MethodInfo) at C:\Users\dabarbet\source\repos\roslyn\src\Compilers\Core\Portable\InternalUtilities\ReflectionUtilities.cs : 115
  + Microsoft.CodeAnalysis.Scripting.ScriptBuilder.Build<T>(Microsoft.CodeAnalysis.Compilation, Microsoft.CodeAnalysis.DiagnosticBag, bool, System.Threading.CancellationToken) at C:\Users\dabarbet\source\repos\roslyn\src\Scripting\Core\ScriptBuilder.cs : 166
  + Microsoft.CodeAnalysis.Scripting.ScriptBuilder.CreateExecutor<T>(Microsoft.CodeAnalysis.Scripting.ScriptCompiler, Microsoft.CodeAnalysis.Compilation, bool, System.Threading.CancellationToken) at C:\Users\dabarbet\source\repos\roslyn\src\Scripting\Core\ScriptBuilder.cs : 88
  + Script<T>.GetExecutor(System.Threading.CancellationToken) at C:\Users\dabarbet\source\repos\roslyn\src\Scripting\Core\Script.cs : 365
  + Script<T>.RunAsync(object, Func<System.Exception, bool>, System.Threading.CancellationToken) at C:\Users\dabarbet\source\repos\roslyn\src\Scripting\Core\Script.cs : 469
  + Script<T>.RunAsync(object, System.Threading.CancellationToken) at C:\Users\dabarbet\source\repos\roslyn\src\Scripting\Core\Script.cs : 443

Expected: False
Actual:   True
```

@dotnet/roslyn-compiler could I get some assistance tracking down this issue?  It is easy to repro locally by just checking out this branch.  I haven't found anything obvious after a cursory investigation, but I'm very unfamiliar with this bit.

File: Format.cmd [hunk 1]
```diff
@@ -0,0 +1,2 @@
+@echo off
+powershell -ExecutionPolicy ByPass -NoProfile -command "& """%~dp0eng\format.ps1""" %*"
```
Comment (sharwell) [inline Format.cmd:2]: ❓  Why is this part script of this PR?

File: dotnet-tools.json [hunk 2]
```diff
@@ -14,4 +14,4 @@
       ]
     }
   }
-}
+}
```
Comment (sharwell) [inline dotnet-tools.json:17]: 📝 This file shouldn't have changed

File: eng/format.ps1 [hunk 3]
```diff
@@ -0,0 +1,4 @@
+# Copyright (c) .NET Foundation and contributors. All rights reserved.
+# Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+dotnet tool run dotnet-format $Args
```
Comment (sharwell) [inline eng/format.ps1:4]: ❓ Why is this file part of this pull request?
📝 File is missing the final newline

File: eng/format.sh [hunk 4]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env bash
+# Copyright (c) .NET Foundation and contributors. All rights reserved.
+# Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+dotnet tool run dotnet-format $@
```
Comment (sharwell) [inline eng/format.sh:5]: 📝 File is missing the final newline

File: format.sh [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+  scriptroot="$( cd -P "$( dirname "$source" )" && pwd )"
+  source="$(readlink "$source")"
+
+  # if $source was a relative symlink, we need to resolve it relative to the path where the
+  # symlink file was located
+  [[ $source != /* ]] && source="$scriptroot/$source"
+done
+
+scriptroot="$( cd -P "$( dirname "$source" )" && pwd )"
+"$scriptroot/eng/format.sh" $@
```
Comment (sharwell) [inline format.sh:16]: 📝 File is missing the final newline

File: src/CodeStyle/VisualBasic/CodeFixes/Microsoft.CodeAnalysis.VisualBasic.CodeStyle.Fixes.vbproj [hunk 6]
```diff
@@ -55,6 +55,9 @@
     <ProjectReference Include="..\..\Core\CodeFixes\Microsoft.CodeAnalysis.CodeStyle.Fixes.csproj" />
     <ProjectReference Include="..\Analyzers\Microsoft.CodeAnalysis.VisualBasic.CodeStyle.vbproj" />
   </ItemGroup>
+  <ItemGroup>
+    <Import Include="System.Threading.Tasks" />
```
Comment (sharwell) [inline src/CodeStyle/VisualBasic/CodeFixes/Microsoft.CodeAnalysis.VisualBasic.CodeStyle.Fixes.vbproj:59]: 😕 Why did this change?

File: src/CodeStyle/VisualBasic/Tests/Microsoft.CodeAnalysis.VisualBasic.CodeStyle.UnitTests.vbproj [hunk 7]
```diff
@@ -65,6 +65,7 @@
     <Import Include="Microsoft.CodeAnalysis.VisualBasic.UnitTests" />
     <Import Include="Roslyn.Test.Utilities" />
     <Import Include="System.Threading.Tasks" />
+    <Import Include="System.Xml.Linq" />
```
Comment (sharwell) [inline src/CodeStyle/VisualBasic/Tests/Microsoft.CodeAnalysis.VisualBasic.CodeStyle.UnitTests.vbproj:68]: 😕 Why did this change?

File: src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpWinForms.cs [hunk 8]
```diff
@@ -114,6 +116,7 @@ public void RenameControl()
             VisualStudio.Editor.EditWinFormButtonEvent(buttonName: "SomeButton", eventName: "Click", eventHandlerName: "SomeButtonHandler");
             // Rename the control
             VisualStudio.Editor.EditWinFormButtonProperty(buttonName: "SomeButton", propertyName: "Name", propertyValue: "SomeNewButton");
+            Thread.Sleep(TimeSpan.FromSeconds(20));
```
Comment (sharwell) [inline src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpWinForms.cs:119]: ❗ This should be removed from the PR

```suggestion
```

File: src/VisualStudio/IntegrationTest/TestUtilities/WellKnownProjectTemplates.cs [hunk 9]
```diff
@@ -20,5 +20,15 @@ public static class WellKnownProjectTemplates
...
         public const string CSharpNetCoreUnitTest = "Microsoft.CSharp.NETCore.UnitTest";
         public const string CSharpNetCoreXUnitTest = "Microsoft.CSharp.NETCore.XUnitTest";
+
+        /// <summary>
+        /// The .cs file created by the <see cref="CSharpNetCoreClassLibrary"/> template.
+        /// </summary>
+        public const string CSharpNetCoreClassLibraryClassFileName = "Class1.cs";
+
+        /// <summary>
+        /// The .cs file created by the <see cref="VisualBasicNetCoreClassLibrary"/> template.
```
Comment (sharwell) [inline src/VisualStudio/IntegrationTest/TestUtilities/WellKnownProjectTemplates.cs:30]:
```suggestion
        /// The .vb file created by the <see cref="VisualBasicNetCoreClassLibrary"/> template.
```

**Review by sharwell — CHANGES_REQUESTED** [review]

File: Format.cmd (hunk 1 above)
Comment (JoeRobich) [reply Format.cmd:2]: We are moving to a version of the SDK which contains "dotnet format" as a CLI command. Invoking the dotnet-format local tool is non-intuitive (`dotnet tool run dotnet-format`) and I thought would benefit from these helper scripts.

File: eng/format.ps1 (hunk 3 above)
Comment (JoeRobich) [reply eng/format.ps1:4]: If a final newline is required, then we should update the .editorconfig with `insert_final_newline = true` for .ps1 and .sh files.

File: src/CodeStyle/VisualBasic/CodeFixes/Microsoft.CodeAnalysis.VisualBasic.CodeStyle.Fixes.vbproj (hunk 6 above)
Comment (JoeRobich) [reply src/CodeStyle/VisualBasic/CodeFixes/Microsoft.CodeAnalysis.VisualBasic.CodeStyle.Fixes.vbproj:59]: The default set of imports has been changing between SDKs. I added a set during the move to the Preview 6 SDK and these are now needed moving to the Preview 7 SDK.

File: src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpWinForms.cs (hunk 8 above)
Comment (dibarbet) [reply src/VisualStudio/IntegrationTest/IntegrationTests/CSharp/CSharpWinForms.cs:119]: Woops, left-over from testing.  Will remove

File: Format.cmd (hunk 1 above)
Comment (JoeRobich) [reply Format.cmd]: Will move these to a follow up PR.

File: dotnet-tools.json (hunk 2 above)
Comment (JoeRobich) [reply dotnet-tools.json]: restored

File: src/CodeStyle/VisualBasic/CodeFixes/Microsoft.CodeAnalysis.VisualBasic.CodeStyle.Fixes.vbproj (hunk 6 above)
Comment (JoeRobich) [reply src/CodeStyle/VisualBasic/CodeFixes/Microsoft.CodeAnalysis.VisualBasic.CodeStyle.Fixes.vbproj:59]: Which hopefully will be addressed by https://github.com/dotnet/sdk/issues/19521

**Review by sharwell — APPROVED** [review]

**Comment by JoeRobich** [comment]
The Linux_Single machine tests have passed. Merging this in since we haven't seen issues in the MacOS leg during previous runs and this unblocks CI.

**Pull request merged**